File: angular/coven/src/app/interfaces/Article.ts
import { Author } from "./Author";
import { Category } from "./Category";
import { Sections } from "./Section";
import { WorldMeta } from "./World";

export interface Article {
    id: string;
    title: string;
    template: string;
    isWip: boolean;
    isDraft: boolean;
    state: string;
    passcode: string;
    wordcount: number;
    creationDate: WorldAnvilDateDTO;
    updateDate: WorldAnvilDateDTO;
    publication_date: WorldAnvilDateDTO;
    notification_date: WorldAnvilDateDTO;
    tags: string;
    url: string;
    category: Category; // Make sure to define the Category interface, based on your Category class
    author: Author;
    world: WorldMeta;
    content: string;
    contentParsed: string;
    sections: Sections;
    relations: any; // Replace with a more specific type if you know the structure
    fullRender: string;
}
export interface WorldAnvilDateDTO {
    date: string; // Note: Date type in C# is serialized as a string in JSON
    timezoneType: number;
    timezone: string;
}

export interface WorldArticlesSummary {
    world: WorldMeta;
    term: string;
    offset: string;
    orderBy: string;
    trajectory: string;
    articles: ArticleMeta[];
}
export interface ArticleMeta {
    id: string;
    title: string;
    state: string;
    isWip: boolean;
    isDraft: boolean;
    templateType: string;
    wordcount?: number;
    views?: number;
    likes?: number;
    exerpt: string;
    tags: string;
    adultContent: boolean;
    lastUpdate: WorldAnvilDateDTO;
    url: string;
    author: Author;
    world: WorldMeta;
}

File: Coven/Coven.Logic/Base Types/Article.cs
using Coven.Data.Meta_Objects;
using Coven.Logic.DTO.WorldAnvil;
using Coven.Logic.Meta_Objects;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Coven.Logic.Base_Types
{
    public class Article
    {
        public Guid id { get; set; }
        public string title { get; set; }
        public string template { get; set; }
        public bool isWip { get; set; }
        public bool isDraft { get; set; }
        public string state { get; set; }
        public string passcode { get; set; }
        public int wordcount { get; set; }
        public WorldAnvilDateDTO creationDate { get; set; }
        public WorldAnvilDateDTO updateDate { get; set; }
        public WorldAnvilDateDTO publicationDate { get; set; }
        public WorldAnvilDateDTO notificationDate { get; set; }
        public string tags { get; set; }
        public string url { get; set; }
        public Category category { get; set; }
        /// <summary>
        /// Username may not populate
        /// </summary>
        public Author author { get; set; }
        public WorldMeta world { get; set; }
        public string content { get; set; }
        public string contentParsed { get; set; }
        public object sections { get; set; }
        public object relations  { get; set; }
        public string fullRender { get; set; }

    }
}
